packages/lib/modules/pool/actions/add-liquidity/form/AddLiquidityForm.tsx

```diff
@@ -103,6 +107,16 @@ function AddLiquidityMainForm() {
     setPriceImpact(priceImpactQuery.data)
   }, [priceImpactQuery.data])
 
+  useEffect(() => {
+    if (priceImpactQuery.error) {
+      const error = priceImpactQuery.error as PriceImpactErrorType
+      const balError = error.shortMessage.split(/\r?\n/)[1]
```

Comment:
`shortMessage` could be `undefined` for some errors so it's safer to just use `message`
And I would go for `error.message.includes('BAL#304')` which reads better (and performance is not critical here).

Reply:
updated!